### PR TITLE
Feature/workflow dsl

### PR DESCRIPTION
**Pull request by maltemoeser**
Provides DSL elements App and WorkflowElement in the workflow model layer as well as a scope provider.

### CODE REVIEW

**Comment by maltemoeser** [comment]
Added corresponding change to reference model in PR https://github.com/ps-md2/md2-projects/issues/1

File: de.wwu.md2.framework/src/de/wwu/md2/framework/MD2.xtext [hunk 1]
```diff
@@ -1704,6 +1704,55 @@ EnumTypeParam:
...
 /////////////////////////////////////
 /////////////////////////////////////
+// New Workflow Concept
+/////////////////////////////////////
+/////////////////////////////////////
+
+/*
+ * A workflow allows to construct multiple applications based on WorkflowElements.
+ */
+Workflow:
```
Comment (Dagefoerde) [inline de.wwu.md2.framework/src/de/wwu/md2/framework/MD2.xtext:1714]: I would suggest to allow specification of WorkflowElements and Apps in arbitrary order

File: de.wwu.md2.framework/src/de/wwu/md2/framework/MD2.xtext [hunk 2]
```diff
@@ -1264,54 +1264,54 @@ enum LocationField:
...
- * several Workflows, Workflows can be nested and there
- * is at most one Workflow active.
+ * several ProcessChains, ProcessChains can be nested and there
+ * is at most one ProcessChain active.
  */
-Workflow:
-	'workflow' name = EID '{'
-		workflowSteps += WorkflowStep*
+ProcessChain:
+	'ProcessChain' name = EID '{'
```
Comment (Dagefoerde) [inline de.wwu.md2.framework/src/de/wwu/md2/framework/MD2.xtext]: For reasons of consistency: shouldn't the keyword "ProcessChain" be started with a lowercase p? (Also line 1829)

**Comment by maltemoeser** [comment]
Wrong capitalization also affected another file which should now be fixed in 8277a2172d0b41c47856e62926eb271711c7a9a3.
What I also see remaining (TBD next week):
- [x] Capitalization in general does not look consistent to me (e.g., `entity` is lowercase but something like `TextInput` and also our new `WorkflowElement` is uppercase)
- [x] Rename `wfStep` to `pcStep` (occurrence: https://gist.github.com/maltemoeser/8af63c2f4a94bc822234)
- [x] How do we deal with the occurrence of `workflow` in Android/iOS generators (see: https://gist.github.com/maltemoeser/12e57f0322d07998edf7) as well as their specific projects (md2-lib-android and Md2Library)

**Pull request merged**